From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Choonghoon Park <choong.park@samsung.com>
Date: Wed, 24 Mar 2021 09:44:13 +0900
Subject: ANDROID: sched/core: Move en/dequeue hooks before related callbacks

Vendors want to get en/dequeueing information and update
some vendor-managed data to modifiy DVFS or scheduling behavior.

But in the current hooking positions, vendors get the information
after all behaviors they want to modify are done.

So need to move the hooks before en/dequeue callbacks
to achieve the "true" goals.

Bug: 183543978

Signed-off-by: Choonghoon Park <choong.park@samsung.com>
Change-Id: I12f8e77054d12a855df10ca9d13a52d417343666
---
 kernel/sched/core.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -1603,9 +1603,8 @@ static inline void enqueue_task(struct rq *rq, struct task_struct *p, int flags)
 	}
 
 	uclamp_rq_inc(rq, p);
-	p->sched_class->enqueue_task(rq, p, flags);
-
 	trace_android_rvh_enqueue_task(rq, p, flags);
+	p->sched_class->enqueue_task(rq, p, flags);
 }
 
 static inline void dequeue_task(struct rq *rq, struct task_struct *p, int flags)
@@ -1619,9 +1618,8 @@ static inline void dequeue_task(struct rq *rq, struct task_struct *p, int flags)
 	}
 
 	uclamp_rq_dec(rq, p);
-	p->sched_class->dequeue_task(rq, p, flags);
-
 	trace_android_rvh_dequeue_task(rq, p, flags);
+	p->sched_class->dequeue_task(rq, p, flags);
 }
 
 void activate_task(struct rq *rq, struct task_struct *p, int flags)
